word_counter.rb

```diff
@@ -0,0 +1,27 @@
+class WordCounter
+  def initialize
+  end
+
+  def count(text)
+    parsed = text.downcase.gsub(/(?:[^\w\s\n'\-]|(?<=[^a-z])\-|\-(?=[^a-z]))+/, '').gsub(/[\n\s]+/, ' ').strip
+    words = parsed.split(' ')
+
+    list = {}
+
+    words.each do |word|
+      if list.key?(word)
+        list[word] += 1
+      else
+        list[word] = 1
+      end
```

Comment:
Check the Ruby documentation for the Each. Should be a big hint as to how you can get that down to a single line. Your eg code is pretty close, but you can totally get it down to just one line. 😃